### PR TITLE
Feature/mage 938 - Replica CLI utilities 

### PR DESCRIPTION
**Pull request by cammonro**
This PR includes:
- Data cleanup operations in the `ReplicaManager` class
- Removal of unused and outdated replica method from `ProductHelper` class
- New `StoreNameFetcher` service
- Exposing index settings from `ProductHelper`
- New CLI namespace for store scoped replica management: `algolia:replicas` 
- CLI to manually initiate sync of replicas
- CLI to manually delete replicas (including identifying potentially unused replicas)
- CLI to manually rebuild replica configuration 

Sample interactions through CLI:
image
image
image

### CODE REVIEW

**Review by damcou — APPROVED** [review]
Looks good ! 👋  (with non-blocking comments if you want to merge fast)
I'm a little bit concerned about the ReplicaManager class which begins to be really massive and hard to follow (especially given the number of methods which have pretty similar names).

File: Console/Command/AbstractReplicaCommand.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    public function getCommandName(): string
+    {
+        return 'algolia:replicas:' . $this->getReplicaCommandName();
+    }
+
+    protected function setAreaCode(): void
+    {
+        try {
+            $this->state->setAreaCode(Area::AREA_CRONTAB);
```
[the remark this answers is not in the document]
Comment (cammonro) [reply Console/Command/AbstractReplicaCommand.php:67]: Yup - this was intentional. The crontab area ensures access to necessary resources without having to load the heavier frontend or adminhtml context. It also allows CLIs to be used in cron jobs.

File: Console/Command/ReplicaSyncCommand.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    protected function getStoreArgumentDescription(): string
+    {
+        return 'ID(s) for store(s) to be synced with Algolia (optional), if not specified all stores will be synced';
+    }
+
+    protected function getCommandDescription(): string
+    {
+        return 'Sync configured sorting attributes in Magento to Algolia replica indices';
```
[the remark this answers is not in the document]
Comment (cammonro) [reply Console/Command/ReplicaSyncCommand.php:51]: It is the same function as the admin but this essentially splits out the more useful "rebuild" command for granularity and ease of testing and debugging. 

It might be a real edge case but I also could envision this invoked from another shell script or cron job, perhaps if there were some kind of external service altering the state of the product data or the sorting configuration outside of the Magento admin. In general I could see this and other future CLIs being useful for such a purpose. 

But in most cases, these won't be needed but only when diagnosing or troubleshooting the replica state. I was planning on adding a section in our Algolia docs to provide more guidance here on when you might use these.

File: Console/Command/ReplicaDeleteCommand.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    /**
+     * Deleting unused replica indices is potentially risky, especially if they have enabled query suggestions on their index
+     * Verify with the end user first!
+     *
+     * @param array $unusedReplicas
+     * @return bool
+     */
+    protected function confirmDeleteUnused(array $unusedReplicas): bool
```
[the remark this answers is not in the document]
Comment (cammonro) [reply Console/Command/ReplicaDeleteCommand.php:125]: You know I thought about adding a confirmation here but felt it's trivial to recreate using the `sync` command. It's not a bad idea though to add it just as a sanity check.

**Comment by cammonro** [comment]
> Looks good ! 👋 (with non-blocking comments if you want to merge fast) I'm a little bit concerned about the ReplicaManager class which begins to be really massive and hard to follow (especially given the number of methods which have pretty similar names).

The ReplicaManager does do a lot and it probably could be broken down further by delegating some functions. For now I've implemented as an interface with a very limited set of exposed methods. But this could probably be an opportunity for future refactoring / fine tuning.

File: Console/Command/ReplicaDeleteCommand.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (cammonro) [reply Console/Command/ReplicaDeleteCommand.php:130]: Confirmation added:
image
image

File: Console/Command/ReplicaSyncCommand.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (cammonro) [reply Console/Command/ReplicaSyncCommand.php:51]: Updated to clarify the intent of the `rebuild` vs regular `sync` command
image
